gateway/gateway.go

```diff
@@ -62,6 +63,27 @@ type GetRange struct {
 	To   *int64
 }
 
+type GetResponse struct {
+	bytes     files.File
+	directory *directoryResponse
```

Comment:
Stronger language, so it is clear we don't deal with directory contents here.
```suggestion
	directoryMetadata *directoryMetadata
```